resources/js/src/app/components/pageDesign/LanguageDetection.vue

```diff
@@ -1,62 +1,123 @@
+    <div v-if="targetLang" class="d-flex py-2">
+        <div class="align-self-center mr-auto">{{ textTranslations[targetLang] }}</div>
+
+        <div class="align-self-center text-nowrap">
+            <a :href="redirectUrl" :class="'btn btn-sm btn-appearance'">
+                {{ buttonTranslations[targetLang] }}
+            </a>
+            <a href="#" @click="refuseRedirect()" class="m-sm-1">
+                <i class="fa fa-close"></i>
+            </a>
+        </div>
     </div>
-  </div>
 </template>
 
 <script>
-  export default {
-    data () {
-      return {
-        languageText: '',
-        buttonText: '',
-        languageRedirect: '',
-        textLanguages: ''
-      }
+import { navigateTo } from "../../services/UrlService";
+export default {
+    data()
+    {
+        return {
+            redirectUrl: null,
+            targetLang: null,
+            isMounted: false
+        };
+    },
+    props:
+    {
+        autoRedirect: Boolean,
+        textTranslations: {
+            type: Object,
+            default: () => {}
+        },
+        buttonTranslations: {
+            type: Object,
+            default: () => {}
+        },
+        languageMap: {
+            type: Object,
+            default: () => {}
+        }
+    },
+    created()
+    {
+        // maybe not the smarted solution to prevent CLS
+        this.targetLang = "en";
     },
-    props: ["redirect", "texttranslations", "buttontranslations"],
-    mounted() {
-      this.initializeComponent()
+
+    mounted()
+    {
+        this.isMounted = true;
+
+        if (!App.isShopBuilder)
+        {
+            if (!window.localStorage.getItem('redirectDeactivated'))
+            {
+                this.initialize();
+            }
+            else
+            {
+                this.targetLang = null;
+            }
+        }
     },
+
     methods: {
-      initializeComponent() {
-
-        for (let i = 0; i < window.navigator.languages.length; i++) {
-          const langObject= window.navigator.languages[i].split('-')
-          const linkTag = document.querySelector('link[hreflang="' + langObject[0] +'"]')
-
-          if (App.language !== langObject[0] && linkTag) {
-            if (this.$props.redirect){
-              if (!window.localStorage.getItem('redirectActive')){
-                window.localStorage.setItem('redirectActive', true);
-                window.location.href = linkTag.getAttribute('href')
-              }
-            } else if (!window.localStorage.getItem('redirectDeactivated')) {
-              window.localStorage.removeItem('redirectActive');
-              this.setText(langObject[0]);
-              this.setButton(langObject[0]);
-              this.languageRedirect = linkTag.getAttribute('href')
-              break;
+        initialize()
+        {
+            // exclude the current language from the language list
+            const browserLanguages = this.getBrowserLanguages().filter(
```

Comment:
Den Filter kann man sich glaube ich sparen wenn man das einfach in das erste if der for Schleife packt.